### DISCUSSION TITLE
"Stream" Mod

### DISCUSSION THREAD

**Discussion by Stoppedpuma** [ppy/osu#29410] [Ideas]
The idea behind this mod is to work similarly to the behaviour found in osu!stream's "Stream mode" where the difficulty can go up and down based on the players performance in the map. This could use other difficulties in the beatmap set to make the map easier or harder as mentioned by the players performance.

**Comment by Tenexxt** [comment]
So basically Adaptive Speed but for difficulty? Seems cool tbh

**Comment by airi019** [comment]
how in the world are u going to make that happen with code
sounds impossible

**Reply by Tenexxt** [reply]
I mean, adaptive speed exists, and since for speed mods some stats like AR and OD have to change, they already have that. This time tho, it would work on every stat, and nothing else. Tho i wonder how would CS changing would work, but if we look at some of the mods in the fun category, devs would probably could make it work... the problem is how much work it would require

**Reply by Stoppedpuma** [reply]
As @Tenexxt mentioned, the foundation of changing beatmap statistics dynamically is already there. Besides that, the main part is caching the beatmap objects from all of the difficulties inside of the beatmap set to be changed based on the players performance alongside the statistics of "currently playing" difficulty as well.

**Reply by airi019** [reply]
oh so u mean u want to change to an easier or harder version of same song based on player performance? then it sounds doable. I though u wanted something like random mod on mania and "add notes" in lunatic rave2

**Comment by lamda05** [comment]
Fun fact: [osu!stream is open source now](https://github.com/ppy/osu-stream/blob).

It seems that only HP is affected in osu!stream.

**Reply by Tenexxt** [reply]
This makes everything even easier lol (if devs stick to the orginal source)
